vss-processor-plugin/src/test/kotlin/org/eclipse/kuksa/vssprocessor/plugin/VssProcessorPluginTest.kt

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.eclipse.kuksa.vssprocessor.plugin
+
+import io.kotest.core.spec.style.BehaviorSpec
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.string.shouldContain
+import org.eclipse.kuksa.vssprocessor.plugin.generator.project.AndroidLibProject
+import org.eclipse.kuksa.vssprocessor.plugin.generator.project.VssProcessorLibProject
+import org.eclipse.kuksa.vssprocessor.plugin.generator.project.VssProcessorLibProject.Companion.VSS_TEST_FILE
+import org.eclipse.kuksa.vssprocessor.plugin.generator.project.VssProcessorPluginProject
+import org.gradle.testkit.runner.GradleRunner
+import org.gradle.testkit.runner.TaskOutcome
+import kotlin.io.path.ExperimentalPathApi
+import kotlin.io.path.deleteRecursively
+import kotlin.io.path.exists
+
+@OptIn(ExperimentalPathApi::class)
+class VssProcessorPluginTest : BehaviorSpec({
```

Comment:
From my understanding:

#1: A full clean of the build folder AND gradle cache -> SUCCESS
#2: A full clean of the build folder BUT gradle cache stays -> FROM_CACHE
#3: No clean -> UP_TO_DATE

#2 reduces the execution time because the gradle cache information about the evaluated task (input information etc.) are still there but still executes the task.
#3 Does not execute the task.